hibernate-core/src/main/java/org/hibernate/metamodel/mapping/ModelPartContainer.java

```diff
@@ -30,13 +30,17 @@ default ModelPart findByPath(String path) {
 		int nextStart = 0;
 		int dotIndex;
 		ModelPartContainer modelPartContainer = this;
-		while ( ( dotIndex = path.indexOf( '.', nextStart ) ) != -1 ) {
+		while ( ( dotIndex = path.indexOf( '.', nextStart ) ) != -1   ) {
 			modelPartContainer = (ModelPartContainer) modelPartContainer.findSubPart(
 					path.substring( nextStart, dotIndex ),
 					null
 			);
+			if ( modelPartContainer == null ) {
+				modelPartContainer = this;
```

Comment:
At this point, the method should return null as no model part could be found for the path segment.

Reply:
with nested embedded this was not working